app/src/main/java/au/id/neasbey/biblecast/API/BibleAPI.java

```diff
@@ -0,0 +1,149 @@
+package au.id.neasbey.biblecast.API;
+
+import android.text.Spanned;
+import android.text.TextUtils;
+import android.util.Log;
+
+import java.util.List;
+
+import au.id.neasbey.biblecast.R;
+import au.id.neasbey.biblecast.util.UIUtils;
+
+/**
+ * Created by craigneasbey on 11/08/15.
+ *
+ * Search using the Bible API
+ */
+public abstract class BibleAPI {
+
+    private static final String TAG = BibleAPI.class.getSimpleName();
+
+    private String query;
+
+    private String URL;
+
+    private String auth;
+
+    private String versions;
+
+    private List<Spanned> returnedList;
+
+    private String result;
+
+    public String getQuery() {
+        return query;
+    }
+
+    public void setQuery(String query) {
+        this.query = query;
+    }
+
+    public String getURL() {
+        return URL;
+    }
+
+    public void setURL(String URL) {
+        this.URL = URL;
+    }
+
+    public String getAuth() {
+        return auth;
+    }
+
+    public void setAuth(String auth) {
+        this.auth = auth;
+    }
+
+    public String getVersions() {
+        return versions;
+    }
+
+    public void setVersions(String versions) {
+        this.versions = versions;
+    }
+
+    public List<Spanned> getReturnedList() {
+        return returnedList;
+    }
+
+    protected void setReturnedList(List<Spanned> returnedList) {
+        this.returnedList = returnedList;
+    }
+
+    public String getResult() {
+        return result;
+    }
+
+    protected void setResult(String result) {
+        this.result = result;
+    }
+
+    /**
+     * Performs the search(query) on specified in the URL
+     *
+     * @param bibleAPIResponseHandler Handler for response from the Bible API
+     */
+    public void performRequest(BibleAPIResponseHandler bibleAPIResponseHandler) {
```

Comment:
I am not sure about passing in a BibleAPIResponseHandler here ... should it be injected via the constructor instead?